### PR TITLE
Initial draft of the opent2t cli to interact with onboarding/hub/tran…

### PR DESCRIPTION
**Pull request by arjun-msft**
…slators

Please note this is draft.  Readme still needs more work.

### CODE REVIEW

File: helpers.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+var fs = require('fs');
+var q = require('q');
+var colors = require('colors');
+
+// writes the given array to a set of files
+// one file per array object
+// file name is based on fileNameBase + array[i][propertyToAppend]
+function writeArrayToFile(arrayObject, fileNameBase, propertyToAppend, i) {
+    if (!!!i) {
```
Comment (ChuckFerring-zz) [inline helpers.js]: Why the triple not instead of a single not? #Resolved

File: helpers.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        i = 0;
+    }
+
+    if (i >= arrayObject.length) {
+        return;
+    }
+    
+    var item = arrayObject[i];
```
Comment (ChuckFerring-zz) [inline helpers.js]: Any reason to not use let for local vars? #Resolved

File: helpers.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    console.log("------ Saving device %j to: %j", item[propertyToAppend], fileName); 
+    var contents = JSON.stringify(item);
+    fs.writeFile(fileName, contents, function (err, data) {
+        if (err) {
+            return console.log(err);
+        }
+
+        return writeArrayToFile(arrayObject, fileNameBase, propertyToAppend, i+1);
+    })
```
Comment (ChuckFerring-zz) [inline helpers.js]: Semicolon needed here? #Resolved

File: index.js [hunk 4]
```diff
@@ -6,23 +6,131 @@
 
 var program = require('commander');
 var colors = require('colors');
+var OnboardingCli = require("./onboardingCli");
+var TranslatorCli = require("./translatorCli");
+var fs = require('fs');
+var helpers = require('./helpers');
+var translatorCli = new TranslatorCli();
```
Comment (ChuckFerring-zz) [inline index.js:13]: You have vars called TranslatorCli and translatorCli.  This could be confusing. #ByDesign

File: index.js [hunk 5]
```diff
@@ -6,23 +6,131 @@
...
-    console.log(make_red('Validating lamp schema implementation: NOT IMPLEMENTED YET'));
-} else if (program.validate) {
-    console.log(make_red('Validating lamp schema implementation: NOT IMPLEMENTED YET'));
-} else {
+console.log('');
+
+if (program.test) {
+    // designated to be used for testing out cli scenarios
+}
+
```
Comment (ChuckFerring-zz) [inline index.js]: Is the if(program.test) needed? #Resolved

File: index.js [hunk 6]
```diff
@@ -6,23 +6,131 @@
...
+else if (program.onboarding) {
+    console.log("------ Doing onboarding for %j".header, program.onboarding);
+    
+    var fileName = "./" + program.onboarding + "_onboardingInfo.json";
```
Comment (ChuckFerring-zz) [inline index.js]: This pattern for constructing the fileName is repeated multiple times.  Consider creating a helper function to construct this. #Resolved

File: onboardingCli.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
+/* jshint esversion: 6 */
+/* jshint node: true */
```
Comment (ChuckFerring-zz) [inline onboardingCli.js:2]: Are you actually using jshint? #WontFix

**Comment by ChuckFerring-zz** [comment]
:clock1:

File: index.js (hunk 4 above)
Comment (arjun-msft) [reply index.js:13]: By design, TranslatorCli is the class, translatorCli is the instance. It's a pattern we have been using.  I'm happy to use a different pattern, please recommend one!

---

In reply to: [78631863](https://github.com/openT2T/opent2t-cli/pull/9#discussion_r78631863) [](ancestors = 78631863)

File: onboardingCli.js (hunk 7 above)
Comment (arjun-msft) [reply onboardingCli.js:2]: I don't think so, it was in the base cli that I added to.

---

In reply to: [78635155](https://github.com/openT2T/opent2t-cli/pull/9#discussion_r78635155) [](ancestors = 78635155)

**Review by ChuckFerring-zz — APPROVED** [review]

**Pull request merged**